procrastinate/worker.py

```diff
@@ -388,6 +370,10 @@ async def _poll_jobs_to_abort(self):
                 )
                 # recover from errors and continue polling
 
+    def _handle_abort_jobs_requested(self, job_ids: Iterable[int]):
+        running_job_ids = {c.job.id for c in self._running_jobs.values() if c.job.id}
+        self._job_ids_to_abort.update(running_job_ids.intersection(job_ids))
```

Comment:
`running_job_ids.intersection(job_ids)` can be written as:
```python
self._job_ids_to_abort |= running_job_ids & job_ids
```

Also, I believe it's safer to do:
```python
self._job_ids_to_abort |= job_ids
self._job_ids_to_abort &= running_job_ids
```
(this would effectively remove from _job_ids_to_abort any id that's not a currently running job)